### PR TITLE
unison-ucm: 0.5.19 -> 0.5.20

### PR DESCRIPTION
**Pull request by ereslibre**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/unison/default.nix [hunk 1]
```diff
@@ -31,14 +33,18 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ makeWrapper ]
     ++ lib.optional (!stdenv.isDarwin) autoPatchelfHook;
-  buildInputs = lib.optionals (!stdenv.isDarwin) [ ncurses6 zlib gmp ];
+  buildInputs = lib.optionals (!stdenv.isDarwin) [ gmp ncurses6 zlib ];
```
Comment (sellout) [inline pkgs/development/compilers/unison/default.nix:36]: Heh, nice. This order had bugged me as well 😆

File: pkgs/development/compilers/unison/default.nix [hunk 2]
```diff
@@ -31,14 +33,18 @@ stdenv.mkDerivation (finalAttrs: {
...
   installPhase = ''
     mkdir -p $out/bin
-    mv ucm $out/bin
+    mv runtime $out/runtime
     mv ui $out/ui
-    wrapProgram $out/bin/ucm \
+    mv unison $out/unison
+    makeWrapper $out/unison/unison $out/bin/ucm \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libb2 ]} \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ openssl ]} \
```
Comment (sellout) [inline pkgs/development/compilers/unison/default.nix:45]:
```suggestion
      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libb2 openssl ]} \
```

File: pkgs/development/compilers/unison/default.nix [hunk 3]
```diff
@@ -31,14 +33,18 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ makeWrapper ]
     ++ lib.optional (!stdenv.isDarwin) autoPatchelfHook;
-  buildInputs = lib.optionals (!stdenv.isDarwin) [ ncurses6 zlib gmp ];
+  buildInputs = lib.optionals (!stdenv.isDarwin) [ gmp ncurses6 zlib ];
 
   installPhase = ''
     mkdir -p $out/bin
-    mv ucm $out/bin
+    mv runtime $out/runtime
```
Comment (sellout) [inline pkgs/development/compilers/unison/default.nix:40]: I think conventionally this should go in `$out/lib` or `$out/lib/runtime`.

**Review by sellout — APPROVED** [review]
Just noting that discussion related to this happened on ceedubs/unison-nix#92.

Some minor suggestions, but I think this is good regardless.

Thanks!

File: pkgs/development/compilers/unison/default.nix (hunk 3 above)
Comment (sellout) [reply pkgs/development/compilers/unison/default.nix:40]: Actually, I see [a bunch of derivations actually use `libexec`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20out%2Flibexec&type=code), but the Nixpkgs manual says [put them in `lib`](https://nixos.org/manual/nixpkgs/stable/#outputlib).

File: pkgs/development/compilers/unison/default.nix (hunk 2 above)
Comment (ereslibre) [reply pkgs/development/compilers/unison/default.nix:45]: Oh thanks! I had pending to check if this worked as expected, and it does :)

File: pkgs/development/compilers/unison/default.nix (hunk 3 above)
Comment (ereslibre) [reply pkgs/development/compilers/unison/default.nix:40]: Either `out` or `lib` yeah, let me update it to `lib`.

**Review by sellout — APPROVED** [review]
Looks great! Thanks again!

**Review by sellout — APPROVED** [review]

**Review by ceedubs — APPROVED** [review]
Nice thank you!

**Pull request merged**